### DISCUSSION TITLE
Total vessels and vessels under way

### DISCUSSION THREAD

**Discussion by g7ruh** [jvde-github/AIS-catcher#159] [Ideas]
Jasper,

Thanks for adding the total number of vessels. I think that it works very well at top right of screen. Is it possible to add the number of vessels under way next to the total number? 

Roger

**Comment by jvde-github** [comment]
Will have a think, perhaps hovering or clicking on this total count can give a break down across several dimensions.

Btw I added a new tab activated by (`-N REALTIME on`) that creates a tab with realtime NMEA data. Also in the Android app as soon as it is accepted by Google. Not spectacular but testing ability to get more realtime data to the webclient.

![image](https://github.com/jvde-github/AIS-catcher/assets/52420030/6c8bdb76-dd78-4ca6-b2db-40ab8e1d8609)

**Reply by g7ruh** [reply]
Jasper,

I like the idea of hovering over the Total circle to show more info, Nice! Does not clutter screen unless you need to see the extra data. Much better than a click, which is extra work. Either way I think it will work, juts my preference is hover.

I will test the new tab in the next few days.

thanks,
Roger

**Reply by g7ruh** [reply]
Jasper,

Re the NMEA tab: will it be possible to pause / stop the display to inspect a line.  I mention it here as I said I would test both versions. They work OK.

Roger

**Reply by jvde-github** [reply]
Like that idea. Currently a proof of concept. Thinking about what to do with this and how to visualize. I personally liked the NMEA streaming in the webclient.

**Comment by jvde-github** [comment]
![image](https://github.com/jvde-github/AIS-catcher/assets/52420030/da53645e-db63-4d34-b15b-c28786a3819b)

needs quite a bit of TLC and some more potential in this menu (e.g. full legend etc). Also the styling has some issues still.

**Reply by Taxom** [reply]
Can you please add a new column to the "ships" table with these symbols for sorting? 
![symbols](https://github.com/jvde-github/AIS-catcher/assets/20313019/02005d1f-5677-449d-b3b1-8a9771de2df8)

**Reply by jvde-github** [reply]
Nice idea!

**Comment by Taxom** [comment]
Is it possible to add "Base station" and "Aid to Navigation" separately in this menu?

**Reply by jvde-github** [reply]
That was my thinking as well (plus all ship icons). Need to play around a little bit with the styling to see how to make this fit.

**Comment by g7ruh** [comment]
Jasper, that works for me too, I agree with @Taxom re other items. 

Suggestion: make sure that the total of the separate items adds up to the number in the circle: I can promise that at least one person will query the value, if it does not!! (that excludes me, I have already just made that point from your image above :)   ). Even if there are things you do not display on the pop-up like Coast Guard helicopters

Also if the number is zero for e.g. "Base station" do you display the line or not? Keeps the pop-up smaller if there is a long list of potential items to display. E.G. I see the CG helicopter less than 20% of the time. My view: I think the list (pop-up) should just display items on list and not zero items headings.

Roger

**Comment by jvde-github** [comment]
It is cleaned up now. Will have a think how to take this forward Thanks for the suggestions.

**Comment by jvde-github** [comment]
Easiest and most fun is to show how it is displayed in the map: 
![Screenshot from 2023-09-10 16-19-26](https://github.com/jvde-github/AIS-catcher/assets/52420030/aec0fff7-bde7-44cf-9b33-f363357b73d2)

**Comment by g7ruh** [comment]
Jasper have you included it in the latest edge version?

I am not sure about the experimental fireworks mode: it looks better in dark screen mode. I guess real fireworks look better at night too!!!

I think the fade should be a bit slower taking about twice as long to 'vanish' as it is a bit fast and looks more like a flashing image rather than a gentle fade.

Just a thought.

Roger

**Reply by jvde-github** [reply]
These shipicons in the table? No just toying a bit to see how difficult it is. Might add in the next few weeks.

Might make the animation speed for the fireworks mode a setting. I didn't like longer as it cluttered too much at higher message rates. Is subjective though so a setting might be a nice solution.

**Comment by g7ruh** [comment]
Good idea re animation speed a setting. I had about 400 vessels on screen, just over half were underway when I looked at it and still felt that it was a touch too fast. Real fireworks have a little longer persistence on the retina.

OK re icons in ship table. Looks nice though and helpful in providing extra information in a visual way about vessels, more so as the rest is all numbers and letters, the icons break that up and you can see what the entry is very fast as the eye interprets the icon much faster than a field of numbers or letters. Let me know when you have added it (should you decide to do it) and I will have a look.

But, a thought: if you add icons, folks may wonder what they are, so you may need a "decode list", maybe as a pop-up when hover over the heading field for that column. If you dont add it, then folks will not wonder what the icons are.

Speaking of which, I note the 4 extra fields in the pop-up of total vessels hover. Nice.

Regards
Roger

**Comment by jvde-github** [comment]
I added something in the Edge, see below.I still need to fix some formatting, which I will do at some point. It seems to be a nice addition.

![Screenshot from 2023-09-10 21-22-42](https://github.com/jvde-github/AIS-catcher/assets/52420030/51e1bb56-87e4-43bc-b12d-276cd5c03ac6)
